tests/controls/windowtest.cpp

```diff
@@ -459,3 +460,81 @@ TEST_CASE_METHOD(WindowTestCase, "Window::SizerErrors", "[window][sizer][error]"
 
     REQUIRE_NOTHROW( delete child );
 }
+
+TEST_CASE_METHOD(WindowTestCase, "Window::Refresh", "[window]")
+{
+    wxWindow* const parent = m_window;
+    wxWindow* const child1 = new wxWindow(parent, wxID_ANY, wxPoint(10, 20), wxSize(80, 50));
+    wxWindow* const child2 = new wxWindow(parent, wxID_ANY, wxPoint(110, 20), wxSize(80, 50));
+    wxWindow* const child3 = new wxWindow(parent, wxID_ANY, wxPoint(210, 20), wxSize(80, 50));
+
+    m_window->SetSize(300, 100);
+
+#if wxDEBUG_LEVEL
+    // to help see the windows when debugging
+    parent->SetBackgroundColour(*wxBLACK);
+    child1->SetBackgroundColour(*wxBLUE);
+    child2->SetBackgroundColour(*wxRED);
+    child3->SetBackgroundColour(*wxGREEN);
+#endif
+
+    // Notice that using EventCounter here will give incorrect results,
+    // so we have to bind each window to a distinct event handler instead.
+
+    static bool isParentPainted;
+    static bool isChild1Painted;
+    static bool isChild2Painted;
+    static bool isChild3Painted;
```

Comment:
I don't think it's really going to be a problem here, but it's still a bit weird to make these variables static, why not just pass their address to the handler? In fact, by doing this we could define the handler only once instead of doing it 4 times.

I.e. we could do (warning, untested code ahead):
```cpp
const auto setFlagOnPaint = [](wxWindow* win, bool* flag)
{
    win->Bind(wxEVT_PAINT, [=](wxPaintEvent&)
    {
        wxPaintDC dc(win);
        *flag = true;
    });
}; 

setFlagOnPaint(parent, isParentPainted);
setFlagOnPaint(child1, isChild1Painted);
...
```

Wouldn't this work?

Reply:
Indeed, this works and is much more elegant and concise.
Thanks for the suggestion.